### PR TITLE
Azure VM Ingredient

### PR DESCRIPTION
**Pull request by csperbeck**
Azure VM Ingredient creates a windows or linux vm based on the options selected, and stands up either the selected vm. For optimal usage, you would pair with the Network Interface and Storage Ingredients. Please review @whilke @JasonPagel @jasonshamilton

### CODE REVIEW

File: ingredient/ingredient-azure-vm/src/functions.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import {BaseUtility, IngredientManager} from '@azbake/core'
+
+export class AzureVmUtils extends BaseUtility {
+
+    public create_resource_name(): string {
+        let util = IngredientManager.getIngredientFunction("coreutils", this.context);
+        const name = util.create_resource_name("sbn", null, false);
```
Comment (whilke) [inline ingredient/ingredient-azure-vm/src/functions.ts]: I assume you want "vm" here vs "sbn" ?

File: ingredient/ingredient-azure-vm/src/functions.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        return name;
+    }
+
+    /*public async get_endpoint(nsName: string, rg: string | null = null): Promise<string> {
```
Comment (whilke) [inline ingredient/ingredient-azure-vm/src/functions.ts]: should delete all this

File: ingredient/ingredient-azure-vm/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+| vmName | yes | Name for the VM |
+| virtualNetworkName | yes | Virtual Network Name |
+| location | no | Location for all resources |
+| windowsOSVersion | no | Windows OS type to deploy |
+| vmSize | no | VM Sizing |
+
+##Linux Properties
+Set: 
+| property | value |
+| osType | windows |
```
Comment (whilke) [inline ingredient/ingredient-azure-vm/README.md]: assume this should be linux

**Comment by csperbeck** [comment]
Committed fixed and updated the ingredient to removed the `osType` parameter from the templates, and remove the parameter from the `bakeParams` before it is submitted for deployment.

**Comment by csperbeck** [comment]
```
[Tue, 02 Jul 2019 22:36:34 GMT] East US Baking recipe windows-test
[Tue, 02 Jul 2019 22:36:34 GMT] East US : windows-test Azure VM Plugin Logging:
[Tue, 02 Jul 2019 22:36:34 GMT] East US : windows-test adding ingredient type[@azbake/ingredient-azure-vm] source[]
[Tue, 02 Jul 2019 22:36:34 GMT] East US : windows-test validating deployment...
[Tue, 02 Jul 2019 22:36:35 GMT] East US : windows-test starting deployment...
[Tue, 02 Jul 2019 22:38:38 GMT] East US : windows-test deployment finished...
```

**Comment by whilke** [comment]
> Committed fixed and updated the ingredient to removed the `osType` parameter from the templates, and remove the parameter from the `bakeParams` before it is submitted for deployment.

Looks like you just need to resolve a conflict with the event-hub ingredient?

**Comment by csperbeck** [comment]
@whilke the main bake [build](https://hchb.visualstudio.com/HCHB/_build/results?buildId=77834) is failing because there is a non UTF-8 character in that package.json. Will figure out how to get it resolved

**Comment by csperbeck** [comment]
Have to exclude the checkout on PR... one sec

**Comment by whilke** [comment]
> @whilke the main bake [build](https://hchb.visualstudio.com/HCHB/_build/results?buildId=77834) is failing because there is a non UTF-8 character in that package.json. Will figure out how to get it resolved

If your branch has the fixed version I can just force the merge through to correct it

**Pull request merged**